### PR TITLE
Feature/multi task

### PR DESCRIPTION
**Pull request by braceal**

### CODE REVIEW

**Comment by braceal** [comment]
**Updates**
- Added basic multi-task trainer for linear AutoEncoder with RMSD prediction network.
- Using uncertainty weighted loss functions from `https://arxiv.org/pdf/1705.07115.pdf`
- Refactored pydantic config to have base class taking care of wandb config
- Added wandb setup utility function
- dict_to_device helper function to move dictionary of torch.Tensor's to device.
- Small updates to ContactMapDataset: optional values interface update, optional requires_grad for scalars.
